src/util/metadata/vo_bit/mod.rs

```diff
@@ -222,3 +223,23 @@ pub fn is_internal_ptr_from_vo_bit<VM: VMBinding>(
 pub unsafe fn is_vo_addr(addr: Address) -> bool {
     VO_BIT_SIDE_METADATA_SPEC.load::<u8>(addr) != 0
 }
+
+#[cfg(feature = "vm_space")]
+use crate::MMTK;
+
+/// Set VO bit for a VM space object.
+/// Objects in the VM space are allocated/managed by the binding. This functio provides a way for
+/// the binding to set VO bit for an object in the space.
+#[cfg(feature = "vm_space")]
+pub fn set_vo_bit_for_vm_space_object<VM: VMBinding>(
```

Comment:
Currently the `mmtk::util::metadata::vo_bit` module is `pub(crate)`, and `set_vo_bit_for_vm_space_object` is not re-exported. So this method is still inaccessible to the VM binding although it is `pub`.

I suggest we make the `vo_bit` module `pub`, but make every `pub` items inside the `vo_bit` module `pub(crate)`, and make `set_vo_bit_for_vm_space_object` `pub`.  Then the VM binding can call this function by `mmtk::util::metadata::vo_bit::set_vo_bit_for_vm_space_object`.

I intended to do this refactoring anyway, and remove the `is_mmtk_object` feature, and move everything related to `is_mmtk_object` into `vo_bit` itself (see [this issue](https://github.com/mmtk/mmtk-core/issues/804)).

Reply:
But I removed this function `set_vo_bit_for_vm_space_object`. I realized that a binding needs to set all the object metadata to properly initialize an object in the VM space. I added a function `MMTK::initialize_vm_space_object()` which calls `SFT::initialize_object_metadata` on `vm_space`.